### DISCUSSION TITLE
Style Selector for Theme

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#5490] [Q&A] [answered]
is it possible to define a selector for a specific theme?
Example: I would like to define a selector that sets the Backgroud to red and Foreground to white on all buttons when the selected theme is Fluent Theme Dark.

**Comment by maxkatz6** [comment, marked as answer]
There is no any build-in selectors for themes, because Avalonia doesn't have proper theming feature yet (it's only possible to swap resource dictionaries and styles).

But personally I would create some global class that reads styles from App and check for FluentTheme.Mode, and also subscribes on styles collection changes to recalculate value. To bind it with windows "mixin"  can be used - https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/Mixins/PressedMixin.cs.

**Reply by maxkatz6** [reply]
And if you use BehaviorsSDK it's possible to create behavior that subscribes on app-styles changes event and invokes action which can set some specific class on object (something like this https://gist.github.com/maxkatz6/2c765560767f20cf0483be8fac29ff22).

**Reply by workgroupengineering** [reply]
Thanks for the reply. I hope that in the future it will be possible to create a style with a theme selector, because swapping between assets is expensive.
